flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequestExecutorImplTest.java

```diff
@@ -128,12 +131,33 @@ public void testIgnoresInterruptsWhileRunning() throws Exception {
         }
     }
 
-    @Test
-    public void testCanBeClosed() throws IOException {
-        TestRequestDispatcher requestProcessor = new TestRequestDispatcher();
+    @Test(timeout = 10000)
+    public void testCanBeClosed() throws Exception {
+        long checkpointId = 1L;
+        ChannelStateWriteRequestDispatcher processor =
```

Comment:
Why can we no longer use the `TestRequestDispatcher`?

Reply:
Because the TestRequestDispatcher don't call the action and discardAction. It can't reproduce this bug.